cli/remotebazel/remotebazel.go

```diff
@@ -167,27 +169,71 @@ func determineRemote(repo *git.Repository) (*git.Remote, error) {
 }
 
 func determineDefaultBranch(repo *git.Repository) (string, error) {
-	branches, err := repo.Branches()
-	if err != nil {
-		return "", status.UnknownErrorf("could not list branches: %s", err)
+	branch, localErr := localDefaultRef(repo)
+	if localErr == nil {
+		return branch, nil
 	}
 
-	allBranches := make(map[string]struct{})
-	err = branches.ForEach(func(branch *plumbing.Reference) error {
-		allBranches[string(branch.Name())] = struct{}{}
-		return nil
-	})
+	log.Warnf("Could not determine default git branch locally. Finding the default branch on the 'origin' remote.\n")
+	branch, remoteErr := remoteDefaultRef(repo)
+	if remoteErr == nil {
+		return branch, nil
+	}
+
+	return "", status.NotFoundErrorf("could not determine default branch: %s", errors.Join(localErr, remoteErr))
+}
+
+func localDefaultRef(repo *git.Repository) (string, error) {
+	var localErr error
+	// Check for existence of
+	//   refs/heads/main
+	//   refs/heads/master
+	for _, defaultRefName := range []plumbing.ReferenceName{plumbing.Master, plumbing.Main} {
+		if _, err := repo.Reference(defaultRefName, false /*resolved*/); err != nil {
+			localErr = errors.Join(localErr, err)
+			continue
+		}
+		return defaultRefName.String(), nil
+	}
+
+	// Check for existence of
+	//   refs/remotes/origin/HEAD
```

Comment:
Should `refs/remotes/origin/HEAD` be included in this list?

Reply:
Yeah. Usually, that's a "symbolic ref" that points to the actual default branch ref of that remote. On my local machine it's something like this

```
> cat .git/refs/remotes/origin/HEAD
ref: refs/remotes/origin/master
```

So the expectation is `repo.Reference("refs/remotes/origin/HEAD", resolved=true)` would yield the `refs/remotes/origin/master` ref after the symbolic ref is resolved. So it would be more accurate than "guessing" for repositories with both `master` and `main` branches available.

I still haven't looked into why this symbolic ref is not created in GitHub Workflows though.